src/nvlink/wrapper.py

```
@@ -0,0 +1,1173 @@
+            >>> wrapper = NVLinkWrapper()
+            >>> wrapper.nv_set_service_key("5UJC-VRFM-448X-F3V4-4")
+            0
+            >>> wrapper.nv_init()
+        """
+        try:
+            result = self._nvlink.NVSetServiceKey(service_key)
+            if result == NV_RT_SUCCESS:
+                logger.info("Service key set successfully")
+            else:
+                logger.warning("Failed to set service key", error_code=result)
+            return result
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)